CsCore/xUnitTests/src/Plugins/CsCoreXUnitTests/com/csutil/tests/http/CsvParserTests.cs

```diff
@@ -55,7 +57,34 @@ private class Person {
             public int Age { get; set; }
             public string Location { get; set; }
         }
+        
+        [Fact]
+        public async Task ExampleUsage3() {
+            // Made ConvertToJson Test based on ´ExampleUsage1´
 
-    }
+            string sampleCsvData = "Name,Age,Location\nAlice,25,New York\nBob,30,San Francisco\n";
+            using var stream = new MemoryStream(Encoding.Default.GetBytes(sampleCsvData));
+            List<List<string>> parsedData = CsvParser.ReadCsvStream(stream);
+    
+            JArray dataAsJson = new JArray();
+
+            foreach (var row in parsedData)
+            {
+                JArray rowArray = new JArray(row.Select(JToken.FromObject));
+                dataAsJson.Add(rowArray);
+            }
 
+            Assert.Equal(3, dataAsJson.Count); // 2 rows
+            Assert.Equal(3, ((JArray)dataAsJson[0]).Count); // 3 columns
+
+            Assert.Equal("Name", dataAsJson[0][0].ToString());
+            Assert.Equal("Age", dataAsJson[0][1].ToString());
+            Assert.Equal("Location", dataAsJson[0][2].ToString());
+
+            Assert.Equal("Alice", dataAsJson[1][0].ToString());
+            Assert.Equal("25", dataAsJson[1][1].ToString());
+            Assert.Equal("New York", dataAsJson[1][2].ToString());
```

Comment:
Looks good so far, now that the data is in a JArray it would be great if you could parse it also to a MyUser class you create here in the test and use the JsonReader.GetReader().Read<List<MyUser>>(dataAsJson) to parse it into a second structure where you do the same assertions on as here above. I would say this would nicely complete the demonstration how to easy work with CSVs and parse them into custom classes. 

If that works nice feel free to refactor the test by extracting this part into its own method and then moving it into the CsvParser class: 
```
            List<List<string>> parsedData = CsvParser.ReadCsvStream(stream);

            JArray dataAsJson = new JArray();

            foreach (var row in parsedData)
            {
                JArray rowArray = new JArray(row.Select(JToken.FromObject));
                dataAsJson.Add(rowArray);
            }
```